openday_scavenger/api/map/service.py

```diff
@@ -0,0 +1,33 @@
+import json
+
+from sqlalchemy.orm import Session
+
+from openday_scavenger.api.puzzles.service import get_all
+
+from .schemas import MapLocations
+
+__all__ = ("get_map_locations",)
+
+
+def get_map_locations(db_session: Session) -> MapLocations:
+    """
+    Return all puzzle locations.
+
+    Args:
+        db_session (Session): The SQLAlchemy session object.
+
+    Returns:
+        List of map location objects entered as the location of puzzles
+    """
+    # Get "map" puzzle answer which contains the pixel co-ordinates of puzzle locations
+    puzzles = get_all(db_session=db_session, only_active=True)
+
+    location_arrays = [p.location for p in puzzles if p.location]
+
+    locations: MapLocations = []
```

Comment:
I think this should be `locations: list[MapLocations] = []

Reply:
Or maybe not - still looking at this.

Reply:
No. Maybe it should be this:
```
puzzles = get_all(db_session=db_session, only_active=True)
    
location_arrays = [p.location for p in puzzles if p.location]

locations: MapLocations = MapLocations(locations=[])
for l_array in location_arrays:
    puzzle_locations = json.loads(l_array)
    for loc in puzzle_locations:
        locations.locations.append(loc)

return locations
```

Reply:
Except that breaks it. So `shrug`.

Reply:
Updated, can you see if you still see the error?